elixir/phoenix-example/app/lib/appsignal_phoenix_example_web/router.ex

```diff
@@ -21,6 +21,9 @@ defmodule AppsignalPhoenixExampleWeb.Router do
     get "/slow", PageController, :slow
     get "/error", PageController, :error
     get "/finch", PageController, :finch
+    get "/tesla/vanilla", TeslaController, :vanilla
```

Comment:
Can you add a scenario that triggers an error in tesla so we can test the exception handling?

Reply:
I completely forgot to test this case manually (it's tested on unit tests, but that uses a fake event) and I've been trying to create an endpoint for it today, but I have not been able to trigger an exception event.

At first I thought something like `Tesla.get!("http://this.domain.does.not.exist")` would trigger an exception event -- but no, that's internally an _error_, which the Telemetry middleware sends as an error reason in the stop event -- and _then_ it's raised as an exception. (We can't report error reasons from the stop event because they're just that, the reason -- no error type, no backtrace)

So then, [looking at the implementation](https://github.com/elixir-tesla/tesla/blob/9fbf2215b5945d916ab6f7e757e9623bda5a158b/lib/tesla/middleware/telemetry.ex#L98-L108), I thought I could add another middleware, _after_ the Telemetry middleware, that throws an error -- and that would be caught in this try-catch. But I've not been able to make it happen.

Now I'm thinking, does this even matter at all? The Tesla instrumentation only creates spans in instrumented contexts (Phoenix, Oban...) and those contexts will catch an error when it is raised. So maybe this instrumentation (and probably also FInch) should just always treat exception events as stop events? 🤔

Reply:
We don't use any of the tesla environment in your PR, so it may not be necessary to capture it in tesla itself, as you say.

Reply:
Giving up on this; will modify the Tesla and Finch instrumentations to treat exception events as stop events.